src/sentry/runner/commands/backup.py

```diff
@@ -16,9 +16,22 @@
 def import_(src):
     "Imports data from a Sentry export."
 
-    for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
-        if obj.object._meta.app_label not in EXCLUDED_APPS:
-            obj.save()
+    try:
+        with transaction.atomic():
+            for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
+                if obj.object._meta.app_label not in EXCLUDED_APPS:
+                    obj.save()
+    # For all database integrity errors, let's warn users to follow our
+    # recommended backup/restore workflow before reraising exception. Most of
+    # these errors come from restoring on a different version of Sentry or not restoring
+    # on a clean install.
+    except IntegrityError as e:
+        warningText = ">> Are you restoring from a backup of the same version of Sentry?\n>> Are you restoring onto a clean database?\n>> If neither, then open an issue here: https://github.com/getsentry/sentry/issues/new/choose"
```

Comment:
Logic was wrong in my [comment](https://github.com/getsentry/sentry/pull/44705#discussion_r1116386657). You should know better than to trust me. :^P

```suggestion
        warningText = ">> Are you restoring from a backup of the same version of Sentry?\n>> Are you restoring onto a clean database?\n>> If so then this IntegrityError might be our fault, you can open an issue here:\n>> https://github.com/getsentry/sentry/issues/new/choose"
```

Reply:
🤦‍♂️